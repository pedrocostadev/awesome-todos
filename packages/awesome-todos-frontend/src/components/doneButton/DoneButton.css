.done-button {
	display: flex;
	background-color: var(--green);
	border: none;
	margin-right: var(--spacing-s);
	outline: none;
}
