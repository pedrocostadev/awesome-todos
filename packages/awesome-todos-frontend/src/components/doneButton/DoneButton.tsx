import React from 'react';
import DoneIcon from './DoneIcon';
import './DoneButton.css';

interface Props {
  onClick(): void;
  disabled?: boolean;
}

const DoneButton: React.FC<Props> = ({ onClick, disabled }) => {
  return (
    <button className="done-button" onClick={onClick} data-testid="done-todo-button" disabled={disabled}>
      <DoneIcon />
    </button>
  );
};

export default DoneButton;
