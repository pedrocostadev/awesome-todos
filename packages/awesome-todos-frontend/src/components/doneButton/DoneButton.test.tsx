import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import DoneButton from './DoneButton';

describe('<DoneButton />', () => {
  const onClick = jest.fn();
  test('It should render', () => {
    const button = render(<DoneButton onClick={onClick} />);
    expect(button).toMatchSnapshot();
  });
  test('It should call onClick', () => {
    const { getByTestId } = render(<DoneButton onClick={onClick} />);
    const doneTodoButton = getByTestId('done-todo-button');
    fireEvent.click(doneTodoButton);
    expect(onClick).toBeCalledTimes(1);
  });
});
