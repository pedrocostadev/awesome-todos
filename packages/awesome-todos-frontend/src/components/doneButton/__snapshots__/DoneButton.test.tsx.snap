// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DoneButton /> It should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="done-button"
        data-testid="done-todo-button"
      >
        <svg
          height="30"
          viewBox="0 0 24 24"
          width="30"
        >
          <path
            d="M20.285 2l-11.285 11.567-5.286-5.011-3.714 3.716 9 8.728 15-15.285z"
            fill="white"
          />
        </svg>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="done-button"
      data-testid="done-todo-button"
    >
      <svg
        height="30"
        viewBox="0 0 24 24"
        width="30"
      >
        <path
          d="M20.285 2l-11.285 11.567-5.286-5.011-3.714 3.716 9 8.728 15-15.285z"
          fill="white"
        />
      </svg>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
